### PR TITLE
Add project

### PR DESCRIPTION
**Pull request by brinxmat**
- added Project class
- added ResearchProject class
- added class to domain of hasName
- added hasProject with domain Publication
- added hasGrant property with domain ResearchProject and range Grant
- added hasApproval body property with domain ResearchProject and range Grant
- added properties for Grant
  - hasSource
  - hasIdentifier
- added properties for Approval
  - hasApprovalDate
  - hasApprovedBy 
  - hasApprovalStatus
  - hasApplicationCode 

Some lines reformatted

### CODE REVIEW

**Review by janbj — APPROVED** [review]
Very good description.

**Review by axthosarouris — APPROVED** [review]
LGTM

File: ontology.ttl [hunk 1]
```diff
@@ -255,10 +255,10 @@
   rdfs:range xsd:string .
 
 :hasNPISubjectHeading a rdf:Property ;
-        rdfs:subPropertyOf dcterms:subject ;
-        rdfs:comment "A NPI subject heading identifier, see https://npi.nsd.no/dok/bakgrunn2003/Norskvitdisinnst.pdf" ;
-        rdfs:domain :EntityDescription ;
-        rdfs:range xsd:string .
+  rdfs:subPropertyOf dcterms:subject ;
```
Comment (brinxmat) [inline ontology.ttl]: Spaces autoformatted

**Review by janbj — APPROVED** [review]

File: ontology.ttl [hunk 2]
```diff
@@ -278,4 +278,60 @@
   rdfs:domain :Identity ;
   rdfs:range xsd:string .
 
+:Project a rdfs:Class ;
+  rdfs:comment "A Project" .
```
Comment (axthosarouris) [inline ontology.ttl]: What is a project? can there be an rdfs:example ?  Is it a research project for example? Can we put an "e.g. Research Project" or sth else in the comment to give more context?

File: ontology.ttl [hunk 3]
```diff
@@ -278,4 +278,60 @@
...
+  rdfs:domain :Grant ;
+  rdfs:range  xsd:string .
+
+:hasIdentifier a rdf:Property ;
+  rdfs:comment "The identifier foor a Grant" ;
+  rdfs:domain :Grant ;
+  rdfs:range  xsd:string .
+
+:Approval a rdfs:Class ;
+  rdfs:comment "The relation linking a Project with an Approval" ;
```
Comment (axthosarouris) [inline ontology.ttl]: I am confused. a class has range and the range is itself?

File: ontology.ttl [hunk 4]
```diff
@@ -278,4 +278,60 @@
...
+  rdfs:domain :Project ;
+  rdfs:range :Approval .
+
+:hasApproval  a rdf:Property ;
+  rdfs:comment "The grant ID of a Project" ;
```
Comment (axthosarouris) [inline ontology.ttl]: Is the comment correct?

**Review by axthosarouris — CHANGES_REQUESTED** [review]
I think there are some copy-paste errors here.

File: ontology.ttl (hunk 3 above)
Comment (brinxmat) [reply ontology.ttl]: This is an error, classes do not have domain/range. Good catch.

File: ontology.ttl (hunk 2 above)
Comment (brinxmat) [reply ontology.ttl]: Following communication with PO, we want to support multiple types of Project, so I will subclass Project -> ResearchProject to open for other kinds of project

File: ontology.ttl (hunk 4 above)
Comment (brinxmat) [reply ontology.ttl]: No!

File: ontology.ttl [hunk 5]
```diff
@@ -296,6 +296,59 @@
...
+  rdfs:domain :ResearchProject ;
+  rdfs:range :Grant .
+
+:hasSource a rdf:Property ;
+  rdfs:comment "The source of funding for a Grant" ;
+  rdfs:domain :Grant ;
+  rdfs:range  xsd:string .
+
+:Approval a rdfs:Class ;
+  rdfs:comment "The relation linking a Project with an Approval" .
```
Comment (axthosarouris) [inline ontology.ttl]: Comment is wrong?

**Review by axthosarouris — APPROVED** [review]
There is one more mistake but I do not want to lock stuff since I will be partially available today

File: ontology.ttl [hunk 6]
```diff
@@ -296,6 +296,59 @@
...
+:hasProject a rdf:Property ;
+  rdfs:comment "The relationship between an Publication and a Project" ;
+  rdfs:domain :Publication ;
+  rdfs:range :Project, :ResearchProject .
+
+:Grant a rdfs:Class ;
+  rdfs:comment "A Grant from a financing body" .
+
+:hasGrant a rdf:Property ;
+  rdfs:comment "The Grant ID of a Project" ;
```
Comment (janbj) [inline ontology.ttl:315]: Specify Research Project?

**Review by janbj — APPROVED** [review]

**Pull request merged**